concept/ConceptManagerImpl.java

```diff
@@ -21,45 +21,44 @@
+import com.vaticle.typedb.client.api.concept.thing.Relation;
 import com.vaticle.typedb.client.api.concept.type.AttributeType;
 import com.vaticle.typedb.client.api.concept.type.EntityType;
 import com.vaticle.typedb.client.api.concept.type.RelationType;
-import com.vaticle.typedb.client.api.concept.type.ThingType;
 import com.vaticle.typedb.client.common.exception.TypeDBException;
-import com.vaticle.typedb.client.concept.thing.ThingImpl;
+import com.vaticle.typedb.client.concept.thing.AttributeImpl;
+import com.vaticle.typedb.client.concept.thing.EntityImpl;
+import com.vaticle.typedb.client.concept.thing.RelationImpl;
 import com.vaticle.typedb.client.concept.type.AttributeTypeImpl;
 import com.vaticle.typedb.client.concept.type.EntityTypeImpl;
 import com.vaticle.typedb.client.concept.type.RelationTypeImpl;
-import com.vaticle.typedb.client.concept.type.ThingTypeImpl;
-import com.vaticle.typedb.protocol.ConceptProto;
-import com.vaticle.typedb.protocol.TransactionProto;
 import com.vaticle.typeql.lang.common.TypeQLToken;
 
-import java.util.List;
 import javax.annotation.Nullable;
-
-import static com.vaticle.typedb.client.common.rpc.RequestBuilder.ConceptManager.getSchemaExceptionsReq;
-import static com.vaticle.typedb.client.common.rpc.RequestBuilder.ConceptManager.getThingReq;
-import static com.vaticle.typedb.client.common.rpc.RequestBuilder.ConceptManager.getThingTypeReq;
-import static com.vaticle.typedb.client.common.rpc.RequestBuilder.ConceptManager.putAttributeTypeReq;
-import static com.vaticle.typedb.client.common.rpc.RequestBuilder.ConceptManager.putEntityTypeReq;
-import static com.vaticle.typedb.client.common.rpc.RequestBuilder.ConceptManager.putRelationTypeReq;
-import static java.util.stream.Collectors.toList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_attribute;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_attribute_type;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_entity;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_entity_type;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_relation;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_relation_type;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_get_schema_exceptions;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_put_attribute_type;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_put_entity_type;
+import static com.vaticle.typedb.client.jni.typedb_client.concepts_put_relation_type;
+import static com.vaticle.typedb.client.jni.typedb_client.schema_exception_code;
+import static com.vaticle.typedb.client.jni.typedb_client.schema_exception_message;
 
 public final class ConceptManagerImpl implements ConceptManager {
+    public final com.vaticle.typedb.client.jni.Transaction transaction;
```

Comment:
I have a feeling we should try to minimise the exposure of JNI objects as much as possible. Does this field _have_ to be public?